Source/1. Functional Programmingming/js/3_LinearFold.js

```diff
@@ -0,0 +1,25 @@
+﻿(function () {
+    "use strict";
+    if (!Array.prototype.linearFoldFn) {
+        Array.prototype.linearFoldFn = function (callback, initialValue) {
+            if (this == null) {
+                throw new Error('Execute function on undefined object');;
+            }
+            if (Object.prototype.toString.call(this) !== '[object Array]') {
+                throw new TypeError();
+            }
+            if (typeof callback !== 'function') {
+                throw new TypeError(callback + ' is not a function');
+            }
+            if (this.length > 0) {
+                var result = callback(initialValue, this[0], 0, this);
```

Comment:
result variable is defined inside of block, but returned outside of it. Won't work for empty array